### PR TITLE
Super-resolution tutorial

### PR DESCRIPTION
**Pull request by t-bovuk**
Adding super-resolution tutorial, along with two notebook tests and two py scripts that can download training data. Notebooks were tested on python 2.7 and 3.5 and CNTK 2.2 .

### CODE REVIEW

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302A_Evaluation_of_Pretrained_Super-resolution_Models_test.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# ==============================================================================
+
+import os
+import re
+import numpy as np
+
+abs_path = os.path.dirname(os.path.abspath(__file__))
+notebook = os.path.join(abs_path, "..", "..", "..", "..", "Tutorials", "CNTK_302A_Evaluation_of_Pretrained_Super-resolution_Models.ipynb")
+datadir = os.path.join(abs_path, "..", "..", "..", "..", "Examples", "Image", "DataSets", "BerkeleySegmentationDataset")
+
```
Comment (sayanpa) [inline Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302A_Evaluation_of_Pretrained_Super-resolution_Models_test.py]: Add the following:

# Run this on GPU only
notebook_deviceIdsToRun = [0]
notebook_timeoutSeconds = 1800

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import numpy as np
+
+abs_path = os.path.dirname(os.path.abspath(__file__))
+notebook = os.path.join(abs_path, "..", "..", "..", "..", "Tutorials", "CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs.ipynb")
+datadir = os.path.join(abs_path, "..", "..", "..", "..", "Examples", "Image", "DataSets", "BerkeleySegmentationDataset")
+
+# Run this on GPU only
+notebook_deviceIdsToRun = [0]
+
+@pytest.fixture(scope='module')
```
Comment (sayanpa) [inline Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py]: comment this out

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+def clean_data(device_id):
+  # Delete the data folder if we're supposed to run
+  if device_id in notebook_deviceIdsToRun:
+    import subprocess
+    args = ["git", "clean", "-fdx", datadir]
+    subprocess.check_call(args)
+
+@pytest.mark.skipif(not reWeekly.search(os.environ.get('TEST_TAG')),
```
Comment (sayanpa) [inline Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py]: we do want to run this more frequently than a weekly run. Please remove this tag.

File: Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs.ipynb [hunk 4]
```diff
@@ -0,0 +1,1928 @@
...
+     "output_type": "stream",
+     "text": [
+      "Data directory is Data/BerkeleySegmentationDataset\n"
+     ]
+    }
+   ],
+   "source": [
+    "#training configuration\n",
+    "MINIBATCH_SIZE = 64\n",
+    "NUM_MINIBATCHES = 1000 if isFast else 200000\n",
```
Comment (sayanpa) [inline Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs.ipynb]: change the fast mode to 200 iterations. Also, set NUM_MINIBATCHES for the entire notebook here. There are other instances in the notebook that needs to be removed for the isFast flag to be true.

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302A_Evaluation_of_Pretrained_Super-resolution_Models_test.py (hunk 1 above)
Comment (xiaoyongzhu) [reply Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302A_Evaluation_of_Pretrained_Super-resolution_Models_test.py]: notebook_deviceIdsToRun = [0]
notebook_timeoutSeconds = 600

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py (hunk 2 above)
Comment (xiaoyongzhu) [reply Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py]: notebook_deviceIdsToRun = [0]
notebook_timeoutSeconds = 1800

#comment this out
#@pytest.fixture(scope='module')

File: Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py (hunk 3 above)
Comment (xiaoyongzhu) [reply Tests/EndToEndTests/CNTKv2Python/Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs_test.py]: #@pytest.mark.skipif(not reWeekly.search(os.environ.get('TEST_TAG')),

**Comment by KeDengMS** [comment]
Seems super resolution tutorial has been merged.

**Pull request closed without merging**